### ISSUE TITLE
Handle WarningAction SilentlyContinue with Assert-PSRule

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule#428]
**Description of the issue**

Currently when `-WarningAction SilentlyContinue` is used with `Assert-PSRule` warning messages are still reported to the command line.

**To Reproduce**

Steps to reproduce the issue:

```powershell
Assert-PSRule -Module PSRule.Rules.Azure -f .\export02\*.json -WarningAction SilentlyContinue;
```

**Expected behaviour**

Assert-PSRule should honor PowerShell warning preference. 

**Module in use and version:**

- Module: PSRule
- Version: **0.15.0-B2002019**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      6.2.4
PSEdition                      Core
GitCommitId                    6.2.4
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
